plugins/tuya/src/camera.ts

```diff
@@ -0,0 +1,209 @@
+import sdk from '@scrypted/sdk';
+import { TuyaController } from "./main";
+import { TuyaDeviceConfig } from "./tuya/const";
+import { TuyaDevice } from "./tuya/device";
+const { deviceManager, mediaManager, systemManager } = sdk;
+
+export class TuyaCameraLight extends ScryptedDeviceBase implements OnOff {
+    constructor(
+        public camera: TuyaCamera,
+        nativeId: string
+    ) {
+        super(nativeId);
+        this.updateState();
+    }
+
+    async turnOff(): Promise<void> {
+        await this.setLightSwitch(false);
+    }
+
+    async turnOn(): Promise<void> {
+        await this.setLightSwitch(true);
+    }
+
+    private async setLightSwitch(on: boolean) {
+        const camera = this.camera.findCamera();
+        const lightSwitchStatus = TuyaDevice.getLightSwitchStatus(camera);
+
+        if (camera.online && lightSwitchStatus) {
+            await this.camera.controller.api.updateDevice(camera, [
+                {
+                    code: lightSwitchStatus.code,
+                    value: on
+                }
+            ]);
+        }
+    }
+
+    updateState(camera?: TuyaDeviceConfig) {
+        camera = camera || this.camera.findCamera();
+        if (!camera)
+            return;
+
+        this.on = TuyaDevice.getLightSwitchStatus(camera)?.value;
+    }
+}
+
+export class TuyaCamera extends ScryptedDeviceBase implements DeviceProvider, VideoCamera, BinarySensor, MotionSensor, OnOff {
+    cameraLight?: TuyaCameraLight
+    private previousMotion?: any;
+
+    constructor(
+        public controller: TuyaController,
+        nativeId: string,
+        config: TuyaDeviceConfig
+    ) {
+        super(nativeId);
+
+        if (this.interfaces.includes(ScryptedInterface.BinarySensor)) {
+            this.binaryState = false;
+        }
+
+        this.updateState(config);
+    }
+
+    // Camera Light Provider
+
+    getDevice(nativeId: string) {
+        if (!this.cameraLight) {
+            this.cameraLight = new TuyaCameraLight(this, nativeId);
+        }
+
+        return this.cameraLight;
+    }
+
+    // OnOff Status Indicator
+
+    async turnOff(): Promise<void> {
+        this.setStatusIndicator(false);
+    }
+
+    async turnOn(): Promise<void> {
+        this.setStatusIndicator(true);
+    }
+
+    private async setStatusIndicator(on: boolean) {
+        const camera = this.findCamera();
+        const statusIndicator = TuyaDevice.getStatusIndicator(camera);
+
+        if (statusIndicator) {
+            await this.controller.api.updateDevice(camera, [
+                {
+                    code: statusIndicator.code,
+                    value: on
+                }
+            ]);
+        }
+    }
+
+    // VideoCamera
+
+    async getVideoStream(
+        options?: MediaStreamOptions
+    ): Promise<MediaObject> {
+        const vsos = await this.getVideoStreamOptions();
+        const vso = vsos.find(find => find.id === options.id) || vsos[0];
```

Comment:
no need for this search, your video stream options will always be the same, since there's only 1 stream. unifi provides 3 streams.